core/services/nmea_injector/TrafficController.py

```diff
@@ -96,6 +112,10 @@ async def add_sock(self, sock: NMEASocket) -> None:
         else:
             raise UnsupportedSocketKind(f"Got {sock.kind}. Expected one of: {[kind.value for kind in SocketKind]}.")
         self._socks[sock] = server_socket
+        settings_spec = NmeaInjectorSettingsSpecV1.from_spec(sock)
```

Comment:
It would be better to have such method encapsulate over a free function or inside the `NMEASocket` class, the reason behind this is that each settings version will need to implement such function unnecessary, like V1, V2, v3, so every time that we update the settings such method will need to exist.
If we move it to a free function or inside the `NMEASocket` class, the function will only need to exist in a single place and be updated when the settings version change, over duplicating such method over and over for each settings.

Reply:
Agree. Was following the same pattern as the Bridget implementation, but this seems better.